measures/GEBAppliancesPeakPeriodShift/measure.rb

```diff
@@ -0,0 +1,230 @@
+    arg.setDefaultValue(0)
+    args << arg
+
+    arg = OpenStudio::Measure::OSArgument.makeStringArgument('schedules_peak_period_column_names', false)
+    arg.setDisplayName('Schedules: Peak Period Schedule File Column Names')
+    arg.setDescription('Comma-separated list of schedule file column names to shift during the peak period.')
+    args << arg
+
+    return args
+  end
+
+  def get_schedule_file_column_names(model)
+    schedule_file_column_names = []
+    model.getExternalFiles.each do |external_file|
+      external_file_path = external_file.filePath.to_s
+      schedule_file_column_names += CSV.foreach(external_file_path).first
+    end
+    return schedule_file_column_names.uniq.sort
+  end
+
+  # define what happens when the measure is run
+  def run(model, runner, user_arguments)
+    super(model, runner, user_arguments) # Do **NOT** remove this line
+
+    # use the built-in error checking
+    if !runner.validateUserArguments(arguments(model), user_arguments)
+      return false
+    end
+
+    schedules_peak_period = runner.getStringArgumentValue('schedules_peak_period', user_arguments)
+    schedules_peak_period_delay = runner.getIntegerArgumentValue('schedules_peak_period_delay', user_arguments)
+    schedules_peak_period_column_names = runner.getStringArgumentValue('schedules_peak_period_column_names', user_arguments).split(',').map(&:strip)
+
+    schedule_file_column_names_enabled = {}
+    get_schedule_file_column_names(model).each do |schedule_file_column_name|
+      schedule_file_column_names_enabled[schedule_file_column_name] = schedules_peak_period_column_names.include?(schedule_file_column_name)
+    end
+
+    if schedule_file_column_names_enabled.empty? || schedule_file_column_names_enabled.values.all? { |value| value == false }
+      runner.registerAsNotApplicable('Did not select any ScheduleFile objects to shift.')
+      return true
+    end
+
+    begin_hour, end_hour = Schedules.parse_time_range(schedules_peak_period)
+
+    if begin_hour >= end_hour
+      runner.registerError("Specified peak period (#{begin_hour} - #{end_hour}) must be at least one hour long.")
+      return false
+    end
+
+    if ((end_hour - begin_hour) + schedules_peak_period_delay > 12)
+      runner.registerError("Specified peak period (#{begin_hour} - #{end_hour}), plus the delay (#{schedules_peak_period_delay}), must be no longer than 12 hours.")
+      return false
+    end
+
+    # get year
+    yd = model.getYearDescription
+    calendar_year = yd.assumedYear
+    calendar_year = yd.calendarYear.get if yd.calendarYear.is_initialized
+    total_days_in_year = Schedules.NumDaysInYear(calendar_year)
+    sim_start_day = DateTime.new(calendar_year, 1, 1)
+
+    # get steps
+    ts = model.getTimestep
+    ts_per_hour = ts.numberOfTimestepsPerHour
+    steps_in_day = ts_per_hour * 24
+
+    model.getExternalFiles.each do |external_file|
+      external_file_path = external_file.filePath.to_s
```

Comment:
Should probably check the schedule type. You wouldn't want to apply a shift to, e.g., a temperature schedule.

Reply:
The burden is on the user to not provide the name of a temperature schedule.